tests/test_qasm_examples.py

```diff
@@ -1,17 +1,136 @@
+                if clean(line_2):
+                    form_lines.append(clean(line_2) + delim)
+    return form_lines
+
+
+def fix_braces(qasm_lines):
+    """Helper function to place braces on separate lines"""
+    form_lines = []
+    for line_1 in qasm_lines:
+        # if a line contains empty braces, format them separately
+        match = re.search(r"{\s*}", line_1)
+        if match:
+            form_lines.append(line_1.replace(match.group(), "{ }"))
+            continue
+
+        # split at "{" and re-add them as separate list-elements
+        res = line_1.split("{")
+        next_qasm_lines = [clean(em) for el in list(zip(["{"]*len(res), res)) for em in el if clean(em)][1:]
+        for line_2 in next_qasm_lines:
+            # split the new list at "}" and re-add them as separate list-elements
+            res = line_2.split("}")
+            lines = [clean(em) for el in list(zip(["}"]*len(res), res)) for em in el if clean(em)][1:]
+            form_lines.extend(lines)
+
+    #asdad
+    tab = False
+    for i in range(1, len(form_lines) - 1):
+        if tab and form_lines[i] != "}":
+            form_lines[i] = "    " + form_lines[i]
+
+        if form_lines[i] == "{":
+            tab = True
+        elif form_lines[i] == "}":
+            tab = False
+
+    return form_lines
+
+
+def fix_ifs(line):
+    """Helper function to format if-statements correctly"""
+    p = re.compile(r"(if)\s*\(\s*(\w+)\s*(==|!=|>=|<=|<|>)\s*(\w+)\s*\)")
+    match = p.match(line)
+    # if an if-string is found, "build up" a correctly formatted string
+    if match:
+        ifstring = match.group(1) + " (" + match.group(2) + " " + match.group(3) + " " + match.group(4) + ")"
+        return ifstring + line[match.span()[1]:]
+    return line
+
+
+def fix_spaces(line):
+    """Helper function to format spaces before/after paranthesis and after commas correctly"""
+    line = ",".join(re.split(r"\s*,\s*", line))
+    line = "(".join(re.split(r"\s*\(\s*", line))
+    line = ") ".join(re.split(r"\s*\)\s*", line))
+    return clean(line)
+
+
+# for variable names that are used by Python, and cannot be used as SymPy symbols
+invalid_var_names = ["lambda"]
+def fix_math_expr(line):
+    """Helper function to format math expressions correctly"""
+
+    # remove the if-parts from the string (doens't support maths in if-statments)
+    if "if" in line:
+        line = re.split(r"if\s*\(.+?\)\s*", line)[-1]
+
+    p = re.compile(r"\(.+\)")
+    res = p.search(line)
+    invalid_var_names_list = []
+    if res and "=" not in res.group() and res.group() not in ["False", "True"]:
+        p_match = res.group()
+        free_params = re.findall(r"[a-zA-Z]\w*", p_match)
+
+        for i, p in enumerate(free_params):
+            if p in invalid_var_names:
+                free_params[i] = p + "_var"
+                p_match = p_match.replace(p, free_params[i])
+                invalid_var_names_list.append((p, free_params[i]))
+
+        sympy.var(free_params)
```

Comment:
Does this line do anything?

Reply:
Yes, it makes the names in `free_params` symbolic.